Independent document editor and CMS CLUE document editor

scytacki opened the pull request:
This adds a stand alone document editor to CLUE it also adds this document editor as a CMS widget.

It also changes the filenames for the vendor code in the built system. Previously there was one `vendor-main-[hash].js` file generated. Now with multiple entry points (runtime, authoring, doc editor) there are multiple `vendor-[entry points]-[hash].js`. The split of those files should be based on which entry points use which packages in node-modules.

tealefristoe commented on src/cms/init-cms.ts:
```diff
@@ -74,6 +75,7 @@ const cmsConfig: CmsConfig = {
 
 export function initCMS() {
+  CMS.registerWidget("clue", ClueControl);
   CMS.registerWidget("json", JsonControl);
```
Should we remove this since it's no longer used, or leave it in case it's useful to play around with it in the future?

tealefristoe commented on src/initialize-app.tsx:
```diff
@@ -0,0 +1,108 @@
+
+export const appConfig = AppConfigModel.create(appConfigSnapshot);
+
+// TODO: we might as well just return
+// stores instead of an object container stores
+export interface IAppProperties {
+  stores: IStores;
+}
```
Seems like this wouldn't be too difficult to change. Should we just take care of it now?

tealefristoe reviewed (approved):
Seems solid! I pointed out a few cleanup tasks we could do before merging, but I don't think any of them are necessary.

scytacki commented:
I think I fixed the issue we saw during the demo at story hour. The comments explain the issue. The workaround is to remove the meta section from the CMS configuration.

scytacki commented:
@tealefristoe this is ready to review again (assuming the tests pass). I address your comments, added a TODO section to the documentation. 

Another thing to look at is my fix for the issue we saw in the story hour demo. Basically I removed the `meta` section from the CMS config. It didn't seem to be needed. Without it we aren't specifying that that the content should be found at `content.json`, but it seems like that doesn't matter. I guess the CMS will just use whatever JSON file it finds in the folder as the content.

tealefristoe reviewed (approved):
Looks great!
